tycho-plugins/repository-utils/src/main/java/org/jboss/tools/tycho/sitegenerator/CompareWithBaselineMojo.java

```diff
@@ -0,0 +1,128 @@
+package org.jboss.tools.tycho.sitegenerator;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.codehaus.plexus.logging.Logger;
+import org.eclipse.sisu.equinox.EquinoxServiceFactory;
+import org.eclipse.tycho.ReactorProject;
+import org.eclipse.tycho.artifacts.TargetPlatform;
+import org.eclipse.tycho.core.ee.shared.ExecutionEnvironmentConfigurationStub;
+import org.eclipse.tycho.core.osgitools.DefaultReactorProject;
+import org.eclipse.tycho.core.shared.TargetEnvironment;
+import org.eclipse.tycho.osgi.adapters.MavenLoggerAdapter;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolutionResult;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolutionResult.Entry;
+import org.eclipse.tycho.p2.resolver.facade.P2Resolver;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolverFactory;
+import org.eclipse.tycho.p2.target.facade.TargetPlatformConfigurationStub;
+import org.osgi.framework.Version;
+
+@Mojo(defaultPhase=LifecyclePhase.VERIFY, requiresProject=false, name="compare-version-with-baselines")
+public class CompareWithBaselineMojo extends AbstractMojo {
+
+	@Parameter(property="project", readonly=true)
+	private MavenProject project;
+
+	@Parameter(property="baselines", name="baselines")
+	private List<String> baselines;
+
+	@Parameter(property="skip")
+	private boolean skip;
+
+	@Requirement
+	@Component
+	private EquinoxServiceFactory equinox;
+
+	@Component
+	private Logger plexusLogger;
+
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (skip) {
+			getLog().info("Skipped");
+			return;
+		}
+		ReactorProject reactorProject = DefaultReactorProject.adapt(project);
+		Set<?> dependencyMetadata = reactorProject.getDependencyMetadata(true);
+		if (dependencyMetadata == null || dependencyMetadata.isEmpty()) {
+			getLog().info("No p2 artifacts created");
```

Comment:
This gets printed exactly on all the projects that does *not* need to know to have any checks made.

I suggest we either add a verbose="true" to enable this or only have it shown when running with -X.

I also suggest to make the message more informative/explicit, i.e. "No p2 artifacts created in build. Skipping p2 baseline version comparison"